src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -623,14 +696,39 @@ public bool HasSkipLocalsInitAttribute
         internal bool IsAutoPropertyWithGetAccessor
             => IsAutoProperty && _getMethod is object;
 
-        protected bool IsAutoProperty
-            => (_propertyFlags & Flags.IsAutoProperty) != 0;
+        internal bool IsAutoProperty
+        {
+            get
+            {
+                return (_propertyFlags & Flags.IsAutoProperty) != 0;
+            }
+        }
 
         /// <summary>
         /// Backing field for automatically implemented property, or
         /// for a property with an initializer.
         /// </summary>
-        internal SynthesizedBackingFieldSymbol BackingField { get; }
+        internal SynthesizedBackingFieldSymbol BackingField
```

Comment:
>BackingField [](http://example.com/codeflow?start=47&length=12)

From the proposed plan for the scoped down PR: "Specifically let’s try to not modify behavior of the BackingField property. I.e. it should return a symbol only if the symbol was created during property symbol creation. In other words, either the property is an old style auto property, or there is an initializer."

Given that, we should remove ```EnsureAccessorsBinding``` call and adjust the implementation accordingly.

Reply:
@AlekseyTs That way, we'll be returning the `BackingField` for the old style auto properties only. But how can we get the simplest scenario `public int P { get => field; }` to work if we will have a null `BackingField`?

What I mean is we will restore the original behavior for both `IsAutoProperty` and `BackingField`, how can we know anything about `field` keyword in that case?

Reply:
Okay I see now you suggested a new API for that purpose.

Reply:
Fixed